### PR TITLE
Add some tests for Linter by itself.

### PR DESCRIPTION
**Pull request by rictic**
Also set up test scaffolding.

- [x] CHANGELOG.md has not been updated. Internal change, no user-visible changes.

---
This change is [Reviewable](https://reviewable.io/reviews/polymer/polymer-linter/11)

### CODE REVIEW

**Comment by rictic** [comment]
Activated travis. Bumping to see if it notices.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 lib
 node_modules
+yarn.lock
```
Comment (abdonrd) [inline .gitignore:3]: Why ignore `yarn.lock`?
https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all

Comment (rictic) [reply .gitignore:3]: I don't think that this is as open and shut as that blog post makes it out to be.

When there's a choice between taking pain upon ourselves vs pushing pain onto our users, I think the ratio has to be pretty large for it to be net positive for us to push the pain down onto users.

Having travis notify us about upstream changes that break us is pretty nice as well, as it happens right when someone is actually working on the repo, and a failing test is probably going to be easier to debug than an issue as reported by a user.

File: src/test/linter_test.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
```
Comment (justinfagnani) [inline src/test/linter_test.ts]: 2017

File: src/test/linter_test.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+import {assert} from 'chai';
+import * as path from 'path';
+import {Analyzer} from 'polymer-analyzer';
+import {FSUrlLoader} from 'polymer-analyzer/lib/url-loader/fs-url-loader';
+import {WarningPrinter} from 'polymer-analyzer/lib/warning/warning-printer';
+
+import {Linter} from '../linter';
+
+const fixtures_dir = path.join(__dirname, '..', '..', 'test');
```
Comment (justinfagnani) [inline src/test/linter_test.ts:23]: `path.resolve(__dirname, '../../test')`

File: src/test/linter_test.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+suite('Linter', () => {
+  suite('.lint', () => {
+    let analyzer: Analyzer;
+    let warningPrinter: WarningPrinter;
+    setup(() => {
```
Comment (justinfagnani) [inline src/test/linter_test.ts:31]: lets add some newlines and let this section breathe a little

**Review by justinfagnani — APPROVED** [review]

File: src/test/linter_test.ts (hunk 3 above)
Comment (rictic) [reply src/test/linter_test.ts:23]: Will that work on Windows?

File: src/test/linter_test.ts (hunk 4 above)
Comment (rictic) [reply src/test/linter_test.ts:31]: done

File: src/test/linter_test.ts (hunk 2 above)
Comment (rictic) [reply src/test/linter_test.ts]: Done

**Pull request merged**